### ISSUE TITLE
[a11y] ExpansionTile missing expansion details and interaction behavior on web

### ISSUE THREAD

**Issue by math1man** [flutter/flutter#143852]
Internal: b/326218346

### Steps to reproduce

This is basically the same issue as https://github.com/flutter/flutter/issues/120650, but specific to web. I believe that issue was fixed but only for Android and iOS (not sure about other platforms either). When using a screen reader on an ExpansionTile element, it reads as "group" instead of "button" and gives no hint as to how to activate it. On Android and iOS, it reads something like "Double-tap to expand" or "Double-tap to collapse".

I'm able to reproduce the iissue with the code sample below just in dartpad, using VoiceOver on my Mac laptop. Run the sample and navigate to the ExpansionTile element using the screen reader.

### Expected results

Screen reader should read something like "Hello World! button, double tap to expand"

### Actual results

Screen reader reads "Hello World! group"

### Code sample

<details open><summary>Code sample</summary>

```
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      debugShowCheckedModeBanner: false,
      home: Scaffold(
        body: Center(
          child: ExpansionTile(
            title: Text('Hello, World!'),
            children: [
              Text('Item 1'),
              Text('Item 2'),
              Text('Item 3'),
            ],
          ),
        ),
      ),
    );
  }
}
```
</details>

### Screenshots or Video

None

### Logs

N/A

### Flutter Doctor output

N/A

**Comment by yjbanov** [comment]
I wasn't aware of https://github.com/flutter/flutter/pull/124281 that fixed the mobile side. I'm not sure that was a good fix for the issue.

On the web, there's a formal way to mark an expandable widget with expanded/collapsed state using the [aria-expanded](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded) attribute. However, Flutter's fix is to pass a piece of text through the `hint` property. We should have a pair of `SemanticsFlag.hasExpansionState` and `SemanticsFlag.isExpanded`, similar to `hasEnabledState`/`isEnabled`. On the web we would translate it to `aria-expanded`. Similarly, on macOS we could use [NSAccessibility.Attribute.expanded](https://developer.apple.com/documentation/appkit/nsaccessibility/attribute/1528260-expanded). Platforms that do not support this attribute directly would use the best fallback method. Unfortunately, we seem to have gone straight with the fallback method at the framework level, so the engine is unable to implement it through a formal a11y API.

As a short-term solution, we can pretend that the web doesn't have `aria-expanded` and implement the hint-based fallback.

Having said that, https://github.com/flutter/flutter/pull/124281 did partially address the issue on the web as well. The widget is interactive using the screen reader, and the accessibility announcement "Expanded"/"Collapsed" is voiced correctly. So I'm going to reduce the severity of the issue to P1.

**Comment by math1man** [comment]
The more pressing issue with this bug is that the widget is not being announced as interactable. That seems like a P0 issue to me.

**Comment by yjbanov** [comment]
Gotcha. To give it the formal "button" role the framework should set the `isButton` field on the semantics node. It looks like in https://github.com/flutter/flutter/pull/124281 it was decided that this widget should not be classified as a button (I don't see `button: true` set in the `Semantics` constructor invocation).

However, with https://github.com/flutter/engine/pull/51529, this is what I get from VoiceOver:

![Screenshot 2024-03-19 at 11 36 47 AM](https://github.com/flutter/flutter/assets/211513/f0f2eaf6-8a8a-4dc4-b666-0e0ca1649e43)

**Comment by math1man** [comment]
That seems reasonable, thanks!

**Comment by yjbanov** [comment]
I filed a follow-up issue here: https://github.com/flutter/flutter/issues/145434
